### PR TITLE
feat(metadata-service): implement fake metadata backend in go (DSP-1579)

### PR DESCRIPTION
**Pull request by BalduinLandolt**
resolves DSP-1579

Not sure if this is actually better than `json-server` with `db.json`...  
This is a simple metadata server implemented in go, that emulates the functionality we used from `json-server`, provides the same API, but rather than one  `db.json`, it gets the data from one JSON file per project.

### CODE REVIEW

**Comment by subotic** [comment]
looks good. I think what is missing (in contrast to json-server) is simple search and serving of the frontend. I'm not sure about search, if @mpro7 used json-server or implemented it directly inside Scelte.

**Comment by BalduinLandolt** [comment]
simple search (very basic, though) is already in there

**Comment by BalduinLandolt** [comment]
> I think what is missing (in contrast to json-server) is [...] serving of the frontend

@subotic could you elaborate how you mean that?

**Comment by mpro7** [comment]
@BalduinLandolt I'm not sure about the file names, should we keep the project names as name of the files or swap with the shortcode?

File: services/metadata/backend/fake-backend/fake-backend.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+	w.Header().Set("Content-Type", "application/json")
+	w.Header().Set("Access-Control-Expose-Headers", "X-Total-Count")
+	// TODO: are any of those needed? json-server has them
+	// w.Header().Set("Content-Type", "charset=utf-8")
+	// w.Header().Set("Cache-Control", "no-cache")
+	// w.Header().Set("Expires", "-1")
+	// w.Header().Set("Pragma", "no-cache")
+	// w.Header().Add("X-Total-Count", "10")
+	// TODO: do we need links to previous and next and first an last?
```
Comment (mpro7) [inline services/metadata/backend/fake-backend/fake-backend.go:120]: FE pagination uses only first and last but if we have all options here enabled, FE can add it too.

Comment (BalduinLandolt) [reply services/metadata/backend/fake-backend/fake-backend.go:120]: Ok, then I'll add them

**Comment by subotic** [comment]
> > I think what is missing (in contrast to json-server) is [...] serving of the frontend
> 
> @subotic could you elaborate how you mean that?

the Svelte frontend needs to be packaged and served by this backend. Marcin can show you what I mean.

**Review by mpro7 — APPROVED** [review]

**Pull request merged**